office365_connector.py

```diff
@@ -486,30 +424,108 @@ def load_state(self):
+        Encrypt and save the current state dictionary to the state file.
 
         :param state: state dictionary
         :return: status
         """
-        try:
-            state = _encrypt_state(state, self._asset_id)
-        except Exception as e:
-            self._dump_error_log(e)
-            self.debug_print("{}: {}".format(MSGOFFICE365_ENCRYPTION_ERROR, str(e)))
-            return phantom.APP_ERROR
+        return super().save_state(self._encrypt_state(state, self._asset_id))
+
+    def _decrypt_state(self, state, salt):
+        """
+        Decrypts the state.
+
+        :param state: state dictionary
+        :param salt: salt used for decryption
+        :return: decrypted state
+        """
+        if not state.get("is_encrypted"):
+            return state
+
+        if state.get("non_admin_auth", {}).get("access_token"):
+            try:
+                state["non_admin_auth"]["access_token"] = encryption_helper.decrypt(
+                    state["non_admin_auth"]["access_token"], salt
+                )
+            except Exception as ex:
+                self.debug_print("{}: {}".format(MSGOFFICE365_DECRYPTION_ERROR,
+                                                              _get_error_msg_from_exception(ex, self)))
+                state["non_admin_auth"]["access_token"] = None
+
+        if state.get("non_admin_auth", {}).get("refresh_token"):
+            try:
+                state["non_admin_auth"]["refresh_token"] = encryption_helper.decrypt(
+                    state["non_admin_auth"]["refresh_token"], salt
+                )
+            except Exception as ex:
+                self.debug_print("{}: {}".format(MSGOFFICE365_DECRYPTION_ERROR,
+                                                              _get_error_msg_from_exception(ex, self)))
+                state["non_admin_auth"]["refresh_token"] = None
+
+        if state.get("admin_auth", {}).get("access_token"):
+            try:
+                state["admin_auth"]["access_token"] = encryption_helper.decrypt(
+                    state["admin_auth"]["access_token"], salt
+                )
+            except Exception as ex:
+                self.debug_print("{}: {}".format(MSGOFFICE365_DECRYPTION_ERROR,
+                                                              _get_error_msg_from_exception(ex, self)))
+                state["admin_auth"]["access_token"] = None
+
+        return state
+
+    def _encrypt_state(self, state, salt):
+        """
+        Encrypts the state.
+
+        :param state: state dictionary
+        :param salt: salt used for encryption
+        :return: encrypted state
+        """
 
-        return super().save_state(state)
+        if state.get("non_admin_auth", {}).get("access_token"):
+            try:
+                state["non_admin_auth"]["access_token"] = encryption_helper.encrypt(
+                    state["non_admin_auth"]["access_token"], salt
+                )
+            except Exception as ex:
```

Comment:
The only thing here that can raise is `encrypt` so what exceptions can it throw and why can't we catch them specifically?

Reply:
We observed that it throws `'ascii' codec can't decode byte` exception when token is corrupted somehow and it does tokens to null in the exception block to generate new tokens from the refresh token in upcoming code executions

Reply:
Then please catch that exception so it's clear what we expect to possibly go wrong

Reply:
We are not sure what exception it raises as it is platform code and it changes gradually I believe it will not harm to add a generic exception, overall we catch the exception and print it so that we can easily traceback.
If any cons to using generic exceptions in this case then please let me know.